### PR TITLE
Adding AQSOL Dataset

### PR DESCRIPTION
**Pull request by rbSparky**
(Related to issue #239)
Reference: [pytorch geometric](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.datasets.AQSOL.html#torch_geometric.datasets.AQSOL)

The AQSOL dataset from the [Benchmarking Graph Neural Networks](http://arxiv.org/abs/2003.00982) paper based on [AqSolDB](https://www.nature.com/articles/s41597-019-0151-1), a standardized database of 9,982 molecular graphs with their aqueous solubility values, collected from 9 different data sources.

Will add tests.

### CODE REVIEW

**Comment by rbSparky** [comment]
Tests to be added

**Comment by rbSparky** [comment]
Works locally with path hardcoded, while downloading it gets stuck on "do you want to download [y/n]" any suggestions?

**Comment by rbSparky** [comment]
For some reason can't see the tests here(the ones that failed aren't loading).
@CarloLucibello do you see any obvious issues that I need to fix here?

**Comment by rbSparky** [comment]
Tests pass, will add more tests and docs

File: src/MLDatasets.jl [hunk 1]
```diff
@@ -165,7 +167,7 @@ function __init__()
     __init__pemsbay()
     __init__temporalbrains()
     __init__windmillenergy()
-
+    __init__aqsol()
```
Comment (CarloLucibello) [inline src/MLDatasets.jl:170]: move up to keep alphabetical ordering

File: src/datasets/graphs/AQSOL.jl [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ 0
+ 1
+ 1
+ ⋮
+ 1
+ 1
+ 1
+
+julia> g.edge_index
+([2, 3], [3, 2])
```
Comment (CarloLucibello) [inline src/datasets/graphs/AQSOL.jl:68]: this doesn't seem the correct edge_index, we should see 42 edges. Show the correct one or don't show anything at all.

File: test/datasets/graphs.jl [hunk 3]
```diff
@@ -278,3 +278,23 @@ end
...
+        @test length(data) == data.metadata["n_observations"]
+        @test length(data.graphs) == split_counts[split]
+
+        for (i, g) in enumerate(data)
+            @test g isa MLDatasets.Graph
+
+            s, t = g.edge_index
+            @test all(1 .<= s .<= g.num_nodes)
+            @test all(1 .<= t .<= g.num_nodes)
+
```
Comment (CarloLucibello) [inline test/datasets/graphs.jl:297]: Let's add the tests
```julia
@test length(s) == g.num_edges
@test length(t) == g.num_edges
```

File: test/datasets/graphs.jl [hunk 4]
```diff
@@ -278,3 +278,23 @@ end
...
+@testset "AQSOL" begin
+    split_counts = Dict(:train => 7985, :val => 998, :test => 999)
+    for split in [:train, :val, :test]
+        data = AQSOL(split=split)
+        @test data isa AbstractDataset
+        @test data.split == split
+        @test length(data) == data.metadata["n_observations"]
+        @test length(data.graphs) == split_counts[split]
+
+        for (i, g) in enumerate(data)
```
Comment (CarloLucibello) [inline test/datasets/graphs.jl:291]: testing every single graph is too slow. We can test a random one with
```julia
g = rand(data)
```
or something similar.

**Comment by CarloLucibello** [comment]
`AQSOL` should be added to the documentation in https://github.com/JuliaML/MLDatasets.jl/blob/master/docs/src/datasets/graphs.md

File: src/datasets/graphs/AQSOL.jl (hunk 2 above)
Comment (rbSparky) [reply src/datasets/graphs/AQSOL.jl]: done

File: test/datasets/graphs.jl (hunk 3 above)
Comment (rbSparky) [reply test/datasets/graphs.jl]: done

**Comment by rbSparky** [comment]
All changes have been made

**Review by CarloLucibello — APPROVED** [review]

**Pull request merged**